### DISCUSSION TITLE
No matter what I do, I cannot save an array of data to a seetings defined as an array.

### DISCUSSION THREAD

**Discussion by FireOpsPhil** [spatie/laravel-settings#267] [Q&A]
Sorry... I posted this as an issue instead of a question.  So let's try this again.  I apologize in advance.  I am completely self taught and just started with PHP/Laravel last year.

I have Laravel-Settings working for a variety of data types, minus arrays. This past week, I tried to add an ARRAY of user ids for the first time. The data is coming to the controller, already formatted as a Json string. If I do a die-dump on the incoming data, it confirms the Json string as follows:
```
array:2 [
0 => "1"
1 => "2"
]
```
In my settings, I have the following:
```
 public array $notifyUsers; 
```
In my controller, I am trying to set the value for that setting as follows:
```
$injurySettings->notifyUsers = $validatedData['notifyUsers'];
```
This causes the following error:
```
Cannot assign **string** to property App\Settings\InjuryModuleSettings::$notifyUsers of type **array**
```
If I change my setting to the following
```
 public string $notifyUsers; 
```
I get the exact opposite error, as if to confirm it is an array:
```
Cannot assign **array** to property App\Settings\InjuryModuleSettings::$notifyUsers of type **string**
```
Is this a bug, or am I missing something?

**Comment by tfevan** [comment]
I got the same problem.

This is happening because when you migrate on your **settings** table the `payload` of `notifyUsers` field was saved as **String**, Ex: `"1, 2"`
From your **settings** table you need to change it to an array as, `["1","2"]`
